DRILL-3894: Upgrade functions MaxDir, MinDir... Optional filename parameter

vdiravka opened the pull request:
Functions MaxDir, MinDir, iMaxDir, iMinDir with one (schema) parameter were added.

parthchandra commented:
I think the preferred way is to have multiple implementations: one with filename as a param and one without (instead of having the parameterAmount label). Take a look at some other implementations (say, lpad, or rpad).
Also, schema is misspelt in the error message.

vdiravka commented:
I used the same approach like in lpad .. functions. You can use Query Directory Functions with filename param or without it.
And for code cleaning I've replaced `MaxDirOnePar` with `MaxDir`, `MaxDirTwoPar` with `MaxDirTwoArg` like in the lpad functions and I've replaced `parameterAmount==2` label with `functionClassName?ends_with("TwoArg")`. 
Also I've added UTest for new Query Directory Functions implementations.

parthchandra commented:
Looks good. +1
